crates/sui-types/src/committee.rs

```diff
@@ -56,21 +60,27 @@ impl Committee {
 
     /// Samples authorities by weight
     pub fn sample(&self) -> &AuthorityName {
-        // Uniform number [0, total_votes) non-inclusive of the upper bound
-        let between = Uniform::from(0..self.total_votes);
-        // OsRng implements CryptoRng and is secure
-        let mut _random = between.sample(&mut OsRng);
-        for (auth, weight) in &self.voting_rights {
-            if *weight > _random {
-                return auth;
-            }
-            _random -= *weight;
-        }
-        unreachable!();
+        // unwrap safe unless committee is empty
+        self.choose_multiple_weighted(1).next().unwrap()
+    }
+
+    fn choose_multiple_weighted(&self, count: usize) -> impl Iterator<Item = &AuthorityName> {
+        self.voting_rights
+            .as_slice()
+            .choose_multiple_weighted(&mut OsRng, count, |(_, weight)| *weight as f64)
+            .unwrap()
```

Comment:
A comment such as "unwrap-safe because we control the stake distribution, which is sanity-checked in `<some location>`,  see [`WeightedError`](https://github.com/rust-random/rand/blob/46e11cd8e5f901a89b26f57c77dc9cf51eb1fcf7/src/distributions/weighted_index.rs#L426-L439)" would be good.

Reply:
done!